### PR TITLE
Adds new hoods to the Surplus Vendors

### PR DESCRIPTION
**Pull request by DrMacCool**
## About The Pull Request

With the addition of the **slick** new ponchos, I realised 'what use is a poncho for your body if you don't have a poncho for your face?'.

This PR adds a new headwear option, the 'USCM Hood', to the surplus vendor. It is purely cosmetic and serves no function besides looking cool and being able to pull it up and down. Though designed to pair specifically with the USCM poncho, it can be worn with anything. 

<details><summary>Screenshots</summary>
<p>
<details><summary>Directional Sprites (Desert)</summary>

d1

de1

ds1

d3

</p>
</details>

<details><summary>Jungle</summary>

j1

j2

</p>
</details>

<details><summary>Snow</summary>

s1

s2

</p>
</details>

<details><summary>Urban</summary>

u2

u1

</p>
</details>

<details><summary>Colonial</summary>

c1

c2

</p>
</details>

</p>
</details>

## Why It's Good For The Game

Marine drip is always good.

## Changelog

:cl:
add: Wearable hoods have been added to the USCM Surplus Vendors for intrepid scouts and brooding anti-heroes alike!

/:cl:

### CODE REVIEW

File: code/game/machinery/vending/vendor_types/squad_prep/squad_prep.dm [hunk 1]
```diff
@@ -281,6 +281,7 @@ GLOBAL_LIST_INIT(cm_vending_clothing_marine, list(
 		list("USCM Poncho", 10, /obj/item/clothing/accessory/poncho, VENDOR_ITEM_REGULAR),
 		list("Firearm Lubricant", 20, /obj/item/prop/helmetgarb/gunoil, VENDOR_ITEM_REGULAR),
 		list("USCM Flair", 20, /obj/item/prop/helmetgarb/flair_uscm, VENDOR_ITEM_REGULAR),
+		list("USCM Hood", 10, /obj/item/clothing/head/hood/, VENDOR_ITEM_REGULAR)
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/squad_prep/squad_prep.dm:284]: no trailing / in path names

File: code/__DEFINES/equipment.dm [hunk 2]
```diff
@@ -142,12 +142,18 @@
...
-//Marine caps only, use for flags_marine_hat.
+//Marine hoodss only, use for flags_marine_hat.
 #define HAT_GARB_OVERLAY (1<<0)
 #define HAT_CAN_FLIP (1<<1)
 #define HAT_FLIPPED (1<<2)
 //===========================================================================================
 
+//===========================================================================================
+//Marine caps only, use for flags_marine_hood.
+#define HAT_GARB_OVERLAY (1<<0)
```
Comment (stanalbatross) [inline code/__DEFINES/equipment.dm:153]: fix your ci

File: code/modules/clothing/head/head.dm [hunk 3]
```diff
@@ -274,6 +274,29 @@
...
+/obj/item/clothing/head/hood/verb/drophood()
+	set name = "Raise/Drop Hood"
+	set category = "Object"
+	set src in usr
+	if(!isliving(usr))
+		return
+	if(usr.is_mob_incapacitated())
+		return
+	if(!(flags_marine_hood & HOOD_CAN_MOVE))
+		to_chat(usr, SPAN_WARNING("[src] can't be moved!"))
```
Comment (stanalbatross) [inline code/modules/clothing/head/head.dm:286]: \The [src]

File: code/modules/clothing/head/head.dm [hunk 4]
```diff
@@ -274,6 +274,29 @@
...
+		return
+
+	flags_marine_hood ^= HOOD_DROPPED
+
+	if(flags_marine_hood & HOOD_DROPPED)
+		to_chat(usr, dropping_message["dropped"])
+		icon_state = base_cap_icon + "_d"
+	else
+		to_chat(usr, dropping_message["raised"])
+		icon_state = base_cap_icon
```
Comment (stanalbatross) [inline code/modules/clothing/head/head.dm:296]: make an update_icon proc and call it here

File: code/modules/clothing/head/head.dm [hunk 5]
```diff
@@ -283,6 +306,29 @@
...
 		)
 
+/obj/item/clothing/head/hood/
+	name = "\improper USCM hood"
+	desc = "A hood issued to USCM marines. Often worn by scouts and snipers navigating harsh conditions. Protects you from the weather, if nothing else"
+	icon_state = "hood"
+	var/list/dropping_message = list(
+		"dropped" = "You drop the hood down, revealing your face.",
+		"raised" = "You raise the hood up, concealing your face."
+		)
```
Comment (stanalbatross) [inline code/modules/clothing/head/head.dm:316]: very shitcode, just make 2 separate vars

File: code/modules/clothing/head/head.dm [hunk 6]
```diff
@@ -283,6 +306,29 @@
 		"unflipped" = "You hook the hat's chinstrap under your chin. Peace of mind is worth a little embarassment."
 		)
 
+/obj/item/clothing/head/hood/
```
Comment (stanalbatross) [inline code/modules/clothing/head/head.dm:309]: no trailing /

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/clothing/head/head.dm (hunk 5 above)
Comment (DrMacCool) [reply code/modules/clothing/head/head.dm:316]: I'm a completely inept coder; just a spriter with some *very basic* literacy. How would you reccommend I redo this?

Comment (DrMacCool) [reply code/modules/clothing/head/head.dm:316]: I think I fixed it. Resolving for now.

File: code/__DEFINES/equipment.dm [hunk 7]
```diff
@@ -142,12 +142,17 @@
 //===========================================================================================
 
 //===========================================================================================
-//Marine caps only, use for flags_marine_hat.
+//Marine capss only, use for flags_marine_hat.
```
Comment (stanalbatross) [inline code/__DEFINES/equipment.dm:145]: ?

File: code/modules/clothing/head/head.dm [hunk 8]
```diff
@@ -283,6 +306,27 @@
 		"unflipped" = "You hook the hat's chinstrap under your chin. Peace of mind is worth a little embarassment."
 		)
 
+/obj/item/clothing/head/hood
+	name = "\improper USCM hood"
+	desc = "A hood issued to USCM marines. Often worn by scouts and snipers navigating harsh conditions. Protects you from the weather, if nothing else"
+	icon_state = "hood"
+	var/dropped = "You drop the hood down, revealing your face."
+	var/raised = "You raise the hood up, concealing your face."
```
Comment (stanalbatross) [inline code/modules/clothing/head/head.dm:314]: move below icon var and call them like dropped_message and raised_message

File: code/modules/clothing/head/head.dm [hunk 9]
```diff
@@ -283,6 +306,27 @@
 		"unflipped" = "You hook the hat's chinstrap under your chin. Peace of mind is worth a little embarassment."
 		)
 
+/obj/item/clothing/head/hood
+	name = "\improper USCM hood"
+	desc = "A hood issued to USCM marines. Often worn by scouts and snipers navigating harsh conditions. Protects you from the weather, if nothing else"
```
Comment (stanalbatross) [inline code/modules/clothing/head/head.dm:311]:
```suggestion
	desc = "A hood issued to USCM marines. Often worn by scouts and snipers navigating harsh conditions. Protects you from the weather, if nothing else."
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/clothing/head/head.dm [hunk 10]
```diff
@@ -274,6 +274,29 @@
...
+/obj/item/clothing/head/hood/verb/drophood()
+	set name = "Raise/Drop Hood"
+	set category = "Object"
+	set src in usr
+	if(!isliving(usr))
+		return
+	if(usr.is_mob_incapacitated())
+		return
+	if(!(flags_marine_hood & HOOD_CAN_MOVE))
+		to_chat(usr, SPAN_WARNING("The [src] can't be moved!"))
```
Comment (stanalbatross) [inline code/modules/clothing/head/head.dm:286]:
```suggestion
		to_chat(usr, SPAN_WARNING("\The [src] can't be moved!"))
```
this backslash is required for text macro to call properly

**Review by morrowwolf — CHANGES_REQUESTED** [review]
If we're going to do hoods let's have them attached to the poncho rather than free floating. This is gonna need some reworking.

**Comment by DrMacCool** [comment]
> If we're going to do hoods let's have them attached to the poncho rather than free floating. This is gonna need some reworking.

Why? I understand the concern, but I don't see why that's necessary. Free-floating hoods have been a piece of clothing since we've been smacking eachother swords and shields. And though I designed the hoods to pair with the ponchos, there are similar items like the sniper's smock and thermal cloak which provide a similar 'aesthetic role'. These are ultimately cosmetics before anything else, so locking it behind an accessory some people might not also want to wear feels pointless, even if it sacrifices some small amount of visual cohesion.

To be completely frank, I can't code for shit - the past god-knows-how-many comments show that. If the need for a rework is that dire, I'm gonna be completely up-front an say I don't think I can do it, and if that's that, that's that.

I realise I'm being argumentative over an absolute non-issue; it's a fucking hoodie in a videogame. You guys have the ultimate say for this, and I'm not trying to overrule that. But, I also don't want what I've made so far completely disregarded over what is ultimately a semantic issue, and not at all critical to it or anything else's balance or functionality.

**Comment by morrowwolf** [comment]
> I realise I'm being argumentative over an absolute non-issue; it's a fucking hoodie in a videogame. You guys have the ultimate say for this, and I'm not trying to overrule that. But, I also don't want what I've made so far completely disregarded over what is ultimately a semantic issue, and not at all critical to it or anything else's balance or functionality.

You can only have so many semantic and non-critical issues be merged until the themes and aesthetic of the game become non-sensical.

**Comment by DrMacCool** [comment]
> > I realise I'm being argumentative over an absolute non-issue; it's a fucking hoodie in a videogame. You guys have the ultimate say for this, and I'm not trying to overrule that. But, I also don't want what I've made so far completely disregarded over what is ultimately a semantic issue, and not at all critical to it or anything else's balance or functionality.
> 
> You can only have so many semantic and non-critical issues be merged until the themes and aesthetic of the game become non-sensical.

If that's your call, I can't debate it. For now, I'll close the PR and go back the the drawing board; I'll see what I can do to salvage it.

**Comment by DrMacCool** [comment]
I've changed my mind. If a second opinion says so, I'll properly close the request.

**Comment by morrowwolf** [comment]
Given ponchos are on their way out this one will also not be making the cut. We appreciate the work you've put into this and we hope to see more contributions in the future. I'm sorry it didn't work out.

**Pull request closed without merging**